overlord/snapstate/handlers_download_test.go

```diff
@@ -161,6 +167,48 @@ func (s *downloadSnapSuite) TestDoDownloadSnapNormal(c *C) {
 	})
 }
 
+func (s *downloadSnapSuite) TestDoDownloadSnapWithDeviceContext(c *C) {
+	s.state.Lock()
+
+	snapstate.ReplaceStore(s.state, nil)
```

Comment:
Maybe add a comment here to make it easier to follow the test? Something like:
```
// unset the store here, it will be initialized from the DeviceContext.CtxStore below to test that the DeviceContext is used
```
(or something similar?)